contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -255,6 +263,78 @@ pub fn set_chain_config(
     }
 }
 
+/// Calculates the destination on token transfer amount.
+///
+/// The amount is calculated based on the token decimals on the source and destination chains.
+/// The calculation is done as following:
+/// 1) `destination_amount` = `source_amount` * 10 ^ (`destination_chain_decimals` - `source_chain_decimals`)
+/// 3) If new_amount is greater than the destination chain's `max_uint`, the translation
+/// fails.
+/// 4) If new_amount is zero, the translation fails.
+fn destination_amount(
```

Comment:
needs some unit tests. can we add unit tests for a variety of inputs to this function? want to make sure this is right.

Reply:
doctest would be the best for this kind of functions, sadly it doesn't work with private fn.